src/pages-and-resources/discussions/DiscussionsSettings.jsx

```diff
@@ -52,7 +52,7 @@ function DiscussionsSettings({ courseId, intl }) {
     } else {
       history.push(discussionsPath);
     }
-  }, [discussionsPath]);
+  }, [isFirstStep]);
```

Comment:
This is a subtle useCallback issue - the dependency array here only depended on `discussionsPath`, but used `isFirstStep` which is derived from `discussionsPath` _and_ `pathname`.  That means that if pathname changes, this function wouldn't be re-evaluated and would still be relying on the old pathname.  This created weird bugs in how the back button worked where it would go to the wrong page depending on which page was loaded first.